opencodeblocks/graphics/blocks/block.py

```diff
@@ -313,7 +278,8 @@ def deserialize(self, data: dict, hashmap: dict = None,
 
         self.setPos(QPointF(*data['position']))
         self.metadata = dict(data['metadata'])
-        self.setTitleGraphics(**self.metadata['title_metadata'])
+        self.title_widget = OCBTitle(
+            data['title'], **self.metadata['title_metadata'])
```

Comment:
Calling a constructor of every step of the resize might have a negative impact on performance as memory has to be allocated on every frame. Consider only changing the widget of the title if possible.